File: pdu/python/geometry_msgs/pdu_cdr_conv_QuaternionStamped.py
from .pdu_pytype_QuaternionStamped import QuaternionStamped
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..std_msgs.pdu_cdr_conv_Header import *
from ..geometry_msgs.pdu_cdr_conv_Quaternion import *
from ..builtin_interfaces.pdu_cdr_conv_Time import *



def py_to_cdr_body_QuaternionStamped(writer: CdrWriter, src: QuaternionStamped):
    py_to_cdr_body_Header(writer, src.header)
    py_to_cdr_body_Quaternion(writer, src.quaternion)


def cdr_body_to_py_QuaternionStamped(reader: CdrReader, dst: QuaternionStamped):
    cdr_body_to_py_Header(reader, dst.header)
    cdr_body_to_py_Quaternion(reader, dst.quaternion)
    return dst


def py_to_cdr_QuaternionStamped(src: QuaternionStamped) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_QuaternionStamped(writer, src)
    return writer.bytes()


def cdr_to_py_QuaternionStamped(cdr_payload) -> QuaternionStamped:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = QuaternionStamped()
    return cdr_body_to_py_QuaternionStamped(reader, dst)

File: pdu/javascript/geometry_msgs/pdu_cdr_conv_QuaternionStamped.js
import { QuaternionStamped } from './pdu_jstype_QuaternionStamped.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Header } from '../std_msgs/pdu_jstype_Header.js';
import { PduHeaderConverter } from '../std_msgs/pdu_cdr_conv_Header.js';
import { Quaternion } from '../geometry_msgs/pdu_jstype_Quaternion.js';
import { PduQuaternionConverter } from '../geometry_msgs/pdu_cdr_conv_Quaternion.js';
import { Time } from '../builtin_interfaces/pdu_jstype_Time.js';
import { PduTimeConverter } from '../builtin_interfaces/pdu_cdr_conv_Time.js';


export class PduQuaternionStampedConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { QuaternionStamped } src
     */
    static to_cdr_body(writer, src) {
        PduHeaderConverter.to_cdr_body(writer, src.header);
        PduQuaternionConverter.to_cdr_body(writer, src.quaternion);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { QuaternionStamped } dst
     * @returns { QuaternionStamped }
     */
    static cdr_body_to_js(reader, dst) {
        PduHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduQuaternionConverter.cdr_body_to_js(reader, dst.quaternion);
        return dst;
    }

    /**
     * @param { QuaternionStamped } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { QuaternionStamped }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new QuaternionStamped());
    }
}
